core/src/main/java/org/springframework/security/access/method/DelegatingAuthorizationManagerBeforeAdvice.java

```diff
@@ -0,0 +1,93 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.access.method;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.aopalliance.intercept.MethodInvocation;
+
+import org.springframework.aop.MethodMatcher;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.aop.support.StaticMethodMatcher;
+import org.springframework.security.authorization.AuthorizationDecision;
+import org.springframework.security.core.Authentication;
+
+/**
+ * An {@link AuthorizationManagerBeforeAdvice} which delegates to a specific
+ * {@link AuthorizationManagerBeforeAdvice} and returns the first non-null
+ * {@link AuthorizationDecision}.
+ *
+ * @author Evgeniy Cheban
+ */
+public final class DelegatingAuthorizationManagerBeforeAdvice
+		implements AuthorizationManagerBeforeAdvice<MethodInvocation> {
+
+	private final MethodMatcher methodMatcher = new StaticMethodMatcher() {
+		@Override
+		public boolean matches(Method method, Class<?> targetClass) {
+			for (AuthorizationManagerBeforeAdvice<MethodAuthorizationContext> delegate : DelegatingAuthorizationManagerBeforeAdvice.this.delegates) {
+				MethodMatcher methodMatcher = delegate.getMethodMatcher();
+				if (methodMatcher.matches(method, targetClass)) {
+					return true;
+				}
+			}
+			return false;
+		}
+	};
+
+	private final List<AuthorizationManagerBeforeAdvice<MethodAuthorizationContext>> delegates;
+
+	/**
+	 * Creates an instance.
+	 * @param delegates the {@link AuthorizationManagerBeforeAdvice}s to use
+	 */
+	public DelegatingAuthorizationManagerBeforeAdvice(
+			List<AuthorizationManagerBeforeAdvice<MethodAuthorizationContext>> delegates) {
+		this.delegates = delegates;
+	}
+
+	@Override
+	public MethodMatcher getMethodMatcher() {
+		return this.methodMatcher;
+	}
+
+	/**
+	 * Delegates to a specific {@link AuthorizationManagerBeforeAdvice} and returns the
+	 * first non-null {@link AuthorizationDecision}.
+	 * @param authentication the {@link Supplier} of the {@link Authentication} to check
+	 * @param methodInvocation the {@link MethodInvocation} to check
+	 * @return an {@link AuthorizationDecision} or null if no
+	 * {@link AuthorizationManagerBeforeAdvice}s could decide
+	 */
+	@Override
+	public AuthorizationDecision check(Supplier<Authentication> authentication, MethodInvocation methodInvocation) {
+		Object target = methodInvocation.getThis();
+		Class<?> targetClass = (target != null) ? AopUtils.getTargetClass(target) : null;
+		MethodAuthorizationContext methodAuthorizationContext = new MethodAuthorizationContext(methodInvocation,
+				targetClass);
+		for (AuthorizationManagerBeforeAdvice<MethodAuthorizationContext> delegate : this.delegates) {
+			AuthorizationDecision decision = delegate.check(authentication, methodAuthorizationContext);
+			if (decision != null) {
```

Comment:
I think this should match the `AffirmativeBased` matcher, since that's the default in `@EnableGlobalMethodSecurity`. It returns the first voter that grants and only denies if one of the voters denied.

Reply:
Done. In the current version the `DelegatingAuthorizationManagerBeforeAdvice` returns the first granted `AuthorizationDecision` and the denied `AuthorizationDecision` only if one of the `DelegatingAuthorizationManagerBeforeAdvice`s denied. If multiple `DelegatingAuthorizationManagerBeforeAdvice`s denies then the first denied `AuthorizationDecision` is returned.